### PR TITLE
B 19897 and b 19898 csr UI and backend int

### PR DESCRIPTION
**Pull request by cameroncaci**
## Summary

Introduces a standalone “Customer Service Representative” (CSR) role that splits from the existing QAE/CSR. This role is to function just like the previous QAE/CSR role with the exception of no longer having permission to create, update, delete, or mark violations on QAE reports found on the move page.

The frontend ticket [B-19898](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A947557) and backend ticket [B-19897](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A947556) were combined into a single PR so that they are tracked via the same build number and can be tested alongside one another for the PO.

**NOTE** The CSR was intentionally not added to the multi-role user because the CSR role has limited permissions. Multi role has all encompassing, meaning the multi-role user cannot successfully replicate the CSR user experience.

### How to test
**Identify a move**
This can be skipped if you already have a desired move ready
Open officelocal
Log in as a multi-role
Identify a “Move approved” status queue item and copy its locator #

**Create a QAE report**
If you are signed in as multi-role, please sign out at this point
Sign back in as a dedicated QAE (Currently QAE/CSR)
Navigate to the identified move by searching for its locator#
Navigate to the “Quality assurance” tab and you should see the ability to create reports
Create a QAE report
image
image

**Review report as a CSR**
Sign out of the QAE role if you’re still signed in
Open officelocal
Select local sign in
Create a dedicated CSR user
image

Navigate to the move you made a report on
You should see the ability to view the report, but not create anything
Verify you can successfully view and download the report
image

**Request CSR account**
Open officelocal, sign out if you are currently signed in
Select "request account'
Fill out the form and select to request a CSR account
It should submit successfully
Log into adminlocal
Approve the requested CSR account
Attempt to sign into officelocal with said account

### CODE REVIEW

File: pkg/handlers/authentication/devlocal.go [hunk 1]
```diff
@@ -33,6 +33,8 @@ const (
 	PrimeSimulatorOfficeUserType string = "Prime Simulator office"
 	// QaeCsrOfficeUserType is a type of user for an Office user
 	QaeCsrOfficeUserType string = "QAE/CSR office"
+	// QaeCsrOfficeUserType is the Customer Service Representative type of user for an Office user
```
Comment (danieljordan-caci) [inline pkg/handlers/authentication/devlocal.go:36]: Copy/paste typo here in your comment. I'm assuming `QaeCsrOfficeUserType` is supposed to be `CustomerServiceRepresentativeOfficeUserType` here

File: pkg/handlers/authentication/devlocal.go [hunk 2]
```diff
@@ -830,6 +843,78 @@ func createUser(h devlocalAuthHandler, w http.ResponseWriter, r *http.Request) (
...
+		}
+
+		office := models.TransportationOffice{
+			Name:      "Truss",
+			AddressID: address.ID,
+			Latitude:  37.7678355,
+			Longitude: -122.4199298,
+			Hours:     models.StringPointer("0900-1800 Mon-Sat"),
+			Gbloc:     gbloc,
+		}
```
Comment (danieljordan-caci) [inline pkg/handlers/authentication/devlocal.go:896]: What're the chances you can change this to anything other than `Truss`? Why are we even still using that name in the app haha

File: pkg/testdatagen/scenario/shared.go [hunk 3]
```diff
@@ -8624,6 +8624,50 @@ func createQaeCsr(appCtx appcontext.AppContext) {
...
+	officeUserExists, err := db.Where("email = $1", email).Exists(&officeUser)
+	if err != nil {
+		log.Panic(fmt.Errorf("Failed to query OfficeUser in the DB: %w", err))
+	}
+	// no need to create
+	if officeUserExists {
+		return
+	}
+
+	/* A user with tio role */
```
Comment (danieljordan-caci) [inline pkg/testdatagen/scenario/shared.go:8640]: Typo in your comment here

File: pkg/testdatagen/scenario/shared.go [hunk 4]
```diff
@@ -8624,6 +8624,50 @@ func createQaeCsr(appCtx appcontext.AppContext) {
...
+	customerServiceRepresentativeRole := roles.Role{}
+	err = db.Where("role_type = $1", roles.RoleTypeCustomerServiceRepresentative).First(&customerServiceRepresentativeRole)
+	if err != nil {
+		log.Panic(fmt.Errorf("Failed to find RoleTypeQaeCsr in the DB: %w", err))
+	}
+
+	csrUUID := uuid.Must(uuid.FromString("72432922-BF2E-45DE-8837-1A458F5D1011"))
```
Comment (danieljordan-caci) [inline pkg/testdatagen/scenario/shared.go:8647]: How come you're hard coding a uuid here? Would generating a uuid be better?

File: pkg/handlers/authentication/devlocal.go [hunk 5]
```diff
@@ -257,6 +261,15 @@ func (h UserListHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
 
+				<form method="post" action="/devlocal-auth/new">
+				  <p>
+					<input type="hidden" name="gorilla.csrf.Token" value="{{.CsrfToken}}">
+					<input type="hidden" name="userType" value="{{.CustomerServiceRepresentativeOfficeUserType}}">
+					` + gblocSelectHTML + `
+					<button type="submit" data-hook="new-user-login-{{.CustomerServiceRepresentativeOfficeUserType}}">Create a New {{.CustomerServiceRepresentativeOfficeUserType}} User</button>
+				  </p>
+				</form>
+
```
Comment (danieljordan-caci) [inline pkg/handlers/authentication/devlocal.go:272]: Would it be possible for you to abbreviate to CSR? Just kind of... Stands out when selecting a user. Everything else is abbreviated.

Screenshot 2024-05-28 at 8 41 06 AM

File: src/components/form/OfficeAccountRequestFields/OfficeAccountRequestFields.jsx [hunk 6]
```diff
@@ -96,6 +96,12 @@ export const OfficeAccountRequestFields = ({ render }) => {
             name="headquartersCheckBox"
             label="Headquarters"
           />
+          <CheckboxField
+            id="customerSupportRepresentativeCheckBox"
+            data-testid="customerSupportRepresentativeCheckBox"
+            name="customerSupportRepresentativeCheckBox"
+            label="Customer Support Representative"
+          />
```
Comment (danieljordan-caci) [inline src/components/form/OfficeAccountRequestFields/OfficeAccountRequestFields.jsx:104]: A little picky, but can you arrange this to be above `Headquarters` and below `Quality Assurance & Customer Support`?
Screenshot 2024-05-28 at 8 45 27 AM

**Review by danieljordan-caci — CHANGES_REQUESTED** [review]
Nice tests, all good stuff. Just have a couple questions & changes! Feel free to bite back.

**Review by traskowskycaci — COMMENTED** [review]
Running into an issue when trying to approve the office user I requested. Logged in as admin:

image

Hit approve here:
image

See this:
image

Console:
image

Server log:
`2024-05-28T19:47:32.194Z        ERROR   adminapi/requested_office_users.go:284  Error creating okta account     {"git_branch": "B-19888-estimate-price-INT", "git_commit": "b652963472f6d8065fd86867e6d33477c9b17b40", "host": "adminlocal:3000", "milmove_trace_id": "c68719c4-c5f9-4c8f-840c-db401d43a9f9", "session_id": "pICtTk07HQaedbpd8ALCXJuGtHED_sF5YHAk7VKXft8"}
github.com/transcom/mymove/pkg/handlers/adminapi.(*UpdateRequestedOfficeUserHandler).Handle.UpdateRequestedOfficeUserHandler.Handle.func1
        /Users/m.traskowsky_cn/Documents/github/mymove/pkg/handlers/adminapi/requested_office_users.go:284
github.com/transcom/mymove/pkg/handlers.(*Config).AuditableAppContextFromRequestWithErrors.func1
        /Users/m.traskowsky_cn/Documents/github/mymove/pkg/handlers/config.go:145
github.com/transcom/mymove/pkg/appcontext.(*appContext).NewTransaction.func1
        /Users/m.traskowsky_cn/Documents/github/mymove/pkg/appcontext/context.go:67
[POP] 2024/05/28 19:47:32 sql - END Transaction --- (conn=tx-4272483568443215767, tx=4272483568443215767)`

**Comment by danieljordan-caci** [comment]
> Running into an issue when trying to approve the office user I requested. Logged in as admin:
> 
> image
> 
> Hit approve here: image
> 
> See this: image
> 
> Console: image
> 
> Server log: `2024-05-28T19:47:32.194Z ERROR adminapi/requested_office_users.go:284 Error creating okta account {"git_branch": "B-19888-estimate-price-INT", "git_commit": "b652963472f6d8065fd86867e6d33477c9b17b40", "host": "adminlocal:3000", "milmove_trace_id": "c68719c4-c5f9-4c8f-840c-db401d43a9f9", "session_id": "pICtTk07HQaedbpd8ALCXJuGtHED_sF5YHAk7VKXft8"} github.com/transcom/mymove/pkg/handlers/adminapi.(*UpdateRequestedOfficeUserHandler).Handle.UpdateRequestedOfficeUserHandler.Handle.func1 /Users/m.traskowsky_cn/Documents/github/mymove/pkg/handlers/adminapi/requested_office_users.go:284 github.com/transcom/mymove/pkg/handlers.(*Config).AuditableAppContextFromRequestWithErrors.func1 /Users/m.traskowsky_cn/Documents/github/mymove/pkg/handlers/config.go:145 github.com/transcom/mymove/pkg/appcontext.(*appContext).NewTransaction.func1 /Users/m.traskowsky_cn/Documents/github/mymove/pkg/appcontext/context.go:67 [POP] 2024/05/28 19:47:32 sql - END Transaction --- (conn=tx-4272483568443215767, tx=4272483568443215767)`

This error is likely due to the DODID number you're providing already existing in Okta. Quick search in the admin console:
Screenshot 2024-05-28 at 3 47 42 PM

So the error is likely cause you're trying to create another Okta user with a duplicate DODID but error handling sucks in the admin app, so we don't know why.

**Comment by traskowskycaci** [comment]
> So the error is likely cause you're trying to create another Okta user with a duplicate DODID but error handling sucks in the admin app, so we don't know why.

Thank you! Retesting now.

File: pkg/handlers/authentication/devlocal.go (hunk 5 above)
Comment (traskowskycaci) [reply pkg/handlers/authentication/devlocal.go:272]: I wish the capitalization for this list made more sense. Probably out of scope for this story but just something this made me notice!

**Review by traskowskycaci — APPROVED** [review]
Works as described for me! I agree with Daniel's feedback as well.

File: pkg/handlers/authentication/devlocal.go (hunk 1 above)
Comment (cameroncaci) [reply pkg/handlers/authentication/devlocal.go:36]: resolved

File: pkg/handlers/authentication/devlocal.go (hunk 2 above)
Comment (cameroncaci) [reply pkg/handlers/authentication/devlocal.go:896]: resolved

File: pkg/testdatagen/scenario/shared.go (hunk 3 above)
Comment (danieljordan-caci) [reply pkg/testdatagen/scenario/shared.go:8640]: You see this one @cameroncaci ?

Comment (cameroncaci) [reply pkg/testdatagen/scenario/shared.go]: fixed typo

File: pkg/testdatagen/scenario/shared.go (hunk 4 above)
Comment (cameroncaci) [reply pkg/testdatagen/scenario/shared.go:8647]: See `/migrations/app/schema/20240522124339_add_csr_to_roles.up.sql` where the CSR role is created with this UUID. Hard referencing the UUID allows for e2e testing (Or in this case, ensuring that our testdatagen correctly identifies the CSR role primary key)

Comment (danieljordan-caci) [reply pkg/testdatagen/scenario/shared.go:8647]: Thank ya sir

**Review by danieljordan-caci — APPROVED** [review]
LGTM! Thanks for the changes.

**Comment by cameroncaci** [comment]
flaky test failing

**Pull request merged**